internal/services/monitor/monitor_metric_alert_resource.go

```diff
@@ -951,6 +951,9 @@ func resourceMonitorMetricAlertActionHash(input interface{}) int {
 	var buf bytes.Buffer
 	if v, ok := input.(map[string]interface{}); ok {
 		buf.WriteString(fmt.Sprintf("%s-", v["action_group_id"].(string)))
+		if m, ok := v["webhook_properties"].(map[string]interface{}); ok && m != nil {
+			buf.WriteString(fmt.Sprintf("%v-", m))
```

Comment:
to correctly handle that map would we need to use every key value pair as part of the hash?

Reply:
https://go.dev/play/p/zMffu2qIVeL

I have also thought that we would need to manually pass the map to get all values. But then notices that for `%+v` (which prints fields+values for maps) it prints fields in sorted order.

Here is the PR for that, I guess - https://github.com/golang/go/issues/21095

Reply:
sounds good! ty